### PR TITLE
[SYCL] Fix method definition of sub_group_mask::group_ballot

### LINKED ISSUE #8201: group_ballot predicate argument should be defaulted
**Describe the bug**
[sycl_ext_sub_group_mask](https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/supported/sycl_ext_oneapi_sub_group_mask.asciidoc#ballot) defines ballot with `predicate=true`, but the implementation does not default the argument:

`group_ballot` 
https://github.com/intel/llvm/blob/b13561c98d3edac22e79fb7fe49110379da48369/sycl/include/sycl/ext/oneapi/sub_group_mask.hpp#L270

**To Reproduce**
```c++
#include <sycl/sycl.hpp>

int main() {
  sycl::queue Q;
  Q.parallel_for(sycl::nd_range<1>{32, 32}, [=](sycl::nd_item<1> item) {
    auto Mask = sycl::ext::oneapi::group_ballot(item.get_sub_group());
  });
}
```

**Environment (please complete the following information):**

- OS: Linux, Windows
- Target device and vendor: Any
- DPC++ version: b13561c98d3edac22e79fb7fe49110379da48369
- Dependencies version: N/A

**Additional context**
None.

### PR DESCRIPTION
**Pull request by haroon26**
Closes #8201
provided default value for `predicate` argument in `sub_group_mask::group_ballot` definition

### CODE REVIEW

**Comment by haroon26** [comment]
I have checked the issue and found that there will be error to provide default argument to `group_ballot` definition.
If I just provide default argument to declaration of the method, another error generates which force to define method right after declaration.
Here, I want the opinion if we should go with finding a way to provide default argument to `group_ballot`, or should change the documentation accordingly?

**Comment by sergey-semenov** [comment]
@haroon26 If I understand correctly, the reason you get an error is because you're adding a default argument to a friend declaration. Adding a non-friend declaration earlier, outside `sub_group_mask`, and specifying the default argument there should do the trick.

**Comment by haroon26** [comment]
Hello @sergey-semenov! Thanks for your suggestion. But again there is a problem if we define the method before class. The defintion says:
```
template <typename Group>
detail::enable_if_t<std::is_same<std::decay_t<Group>, sub_group>::value,
                    sub_group_mask>
group_ballot(Group g, bool predicate = true);
```
And you can see that in the return type, we are actually using `sub_group`, which will be not declared at that point. So this will also cause error.

**Comment by sergey-semenov** [comment]
That can be resolved by adding a forward declaration of `sub_group_mask` as well.

**Comment by AlexeySachkov** [comment]
@haroon26, could you please also add a test for the change? Just a simple `-fsyntax-only` test which checks compilation would be enough

File: sycl/test/check_device_code/group_ballot.cpp [hunk 1]
```diff
@@ -0,0 +1,10 @@
+// RUN: %clangxx -fsycl -fsycl-device-only -fsyntax-only -Xclang -verify -Xclang -verify-ignore-unexpected=note,warning %s
```
Comment (AlexeySachkov) [inline sycl/test/check_device_code/group_ballot.cpp:1]:
```suggestion
// RUN: %clangxx -fsycl -fsycl-device-only -fsyntax-only -Xclang -verify %s
// expected-no-diagnostics
```

**Review by steffenlarsen — COMMENTED** [review]
LGTM! Thank you, @haroon26 !

**Review by steffenlarsen — APPROVED** [review]

**Pull request merged**